lib/rucio/client/accountclient.py

```diff
@@ -256,6 +257,9 @@ def get_account_limits(self, account: str, rse_expression: str, locality: str) -
         :param locality:       The scope of the account limit. 'local' or 'global'.
         """
 
+        existing_accounts = [account_data['account'] for account_data in self.list_accounts()]
+        if account not in existing_accounts:
+            raise AccountNotFound(f"Account '{account}' does not exist.")
```

Comment:
`list_accounts()`, with no args passed to it, lists all Rucio accounts - this seems unnecessary if we just want to verify that a single account exists.

IMO we should instead call `get_account(account)`, which raises `AccountNotFound` if the account is not found

Reply:
Thanks, that makes sense! I didn't realize we had this method and mistakenly thought list_accounts() was handling some sort of caching.